Check if async nominating already

hewison-chris opened the pull request:
This is related to SCP nomination where we are already in process of
nominating but the checkNominate is called by the timer. The check if we
are already in the process of a nomination was removed when a check that
we only confirm once. This adds back the check.

hewison-chris committed:
Check if async nominating already

This is related to SCP nomination where we are already in process of
nominating but the checkNominate is called by the timer. The check if we
are already in the process of a nomination was removed when a check that
we only confirm once was added. This adds back the check.

hewison-chris commented:
It passed but will  trigger another run to be sure.

AndrejMitrovic commented:
Since nobody seems to be sure whether nominating twice is OK or not, how about we ask the stellar devs whether `SCP.nominate` is re-entrant? I can submit an issue to `stellar-core`.

hewison-chris commented:
> Since nobody seems to be sure whether nominating twice is OK or not, how about we ask the stellar devs whether `SCP.nominate` is re-entrant? I can submit an issue to `stellar-core`.

I don't think we are only nominating twice. For each block we want to create from Genesis start time. Which is getting more into the past all the time.

hewison-chris committed:
Enable mulitiple nominations per height

Now we are preventing calling SCP `nominate` multiple times
simultaneously we should reset the flag to enable multiple nominations
per block still. This is moving the reset of `isNominating` to as soon
as the `nominate` routine returns, rather than at `stopNominating`.

hewison-chris commented:
Closing this PR as I think I found a better solution in another branch. Will create a new PR.
